### PR TITLE
Chore: Bump grafana-azure-sdk-go to v1.12.0 and expose AzureAuthEnabled value in config

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

Exposing `AzureAuthEnabled` in config so in [Prometheus](https://github.com/grafana/grafana/blob/8fea101931b2b82cd42e73f48417651c1ddb66ec/pkg/tsdb/prometheus/client/transport.go#L42) we can reach it. 
I'll create a separate PR for that.

**Why do we need this feature?**

For decoupling/externalizing work of Prometheus

### CODE REVIEW

File: pkg/plugins/config/config.go [hunk 1]
```diff
@@ -26,7 +26,8 @@ type Cfg struct {
 	AWSExternalId           string
 
 	// Azure Cloud settings
-	Azure *azsettings.AzureSettings
+	Azure            *azsettings.AzureSettings
+	AzureAuthEnabled bool
```
Comment (andresmgot) [inline pkg/plugins/config/config.go:30]: just curious why these two are different fields

**Review by andresmgot — COMMENTED** [review]
LGTM, just one question (someone more familiar with this can give the final :+1: )

File: pkg/plugins/config/config.go (hunk 1 above)
Comment (aangelisc) [reply pkg/plugins/config/config.go:30]: I concur, there's no reason we shouldn't add this field to the `azure` section of the config. As long as we clarify that it's specific to Prometheus it should be fine.

Comment (itsmylife) [reply pkg/plugins/config/config.go:30]: For now, it is only for Prometheus. I don't see/find any other place we use it. In the future, if we need it in elsewhere then we can move this in azsettings.

**Comment by itsmylife** [comment]
I think @aangelisc can give me final 👍 :)

**Review by aangelisc — APPROVED** [review]
LGTM 🚀

**Pull request merged**